src/TestInfrastructure/Helpers/MefTestHelpers.cs

```diff
@@ -180,5 +180,27 @@ public static void Compose(Assembly exportingAssembly, params object[] importers
             using var container = new CompositionContainer(catalog);
             container.Compose(batch);
         }
+
+        /// <summary>
+        /// Check if type has creation policy non shared.
+        /// </summary>
+        public static bool IsMefComponentNonShared(Type type)
```

Comment:
If you make the method generic it would be syntactically slightly easier to call e.g.

```
public static void CheckIsSingletonMefComponent<T>()
{
  var customAttributes = type.GetCustomAttributes(typeof(T), true);
  ...
}
```